Dev

UnderSaga opened the pull request.

UnderSaga committed:
first commit

UnderSaga committed:
SecondPage

UnderSaga committed:
feat: cool

UnderSaga committed:
Изменения:
- Добавлена реализация бургер-меню.

UnderSaga committed:
Изменения:
- Рабочие попапы

UnderSaga committed:
Изменения:
- Скорректировано поведение шапки при переходе по страницам.

UnderSaga committed:
Изменения:
- Приведение к божескому виду.

hhhatemeee commented on src/App.jsx:
```diff
@@ -0,0 +1,24 @@
+function App() {
+
+  return (
+    <>
```
Можно без <></>, т.к. тут у тебя всё обернуто в Routes

UnderSaga committed:
Изменения:
- Улучшение визуальной составляющей кода;
- Изменение логики работы попапов;
- Улучшение работы слайдера (всё еще плохо).

UnderSaga committed:
Изменения:
- Исправлена работа бургер-меню.

hhhatemeee commented on src/components/Layout.jsx:
```diff
@@ -0,0 +1,153 @@
+    setOpenMenu(false);
+  }
+
+  function changeHeader() {
+    if (window.location.pathname === "/") {
+      setMainPage(true);
+    }
+    if (window.location.pathname === "/pets") {
+      setMainPage(false);
+    }
+  }
+
```
Вместо window.location можно использовать хук из React-Router-DOM - `const { pathname} = useLocation`, он возвращает объект, в котором присутствует pathname. Так более правильнее. Кроме того тут можно не делать отдельный стейт, а просто держать переменную

`const isMainPage = pathname === '/'`

hhhatemeee commented on src/components/Layout.jsx:
```diff
@@ -0,0 +1,153 @@
+import { useState, useEffect } from "react";
+
+function Layout() {
+  const mainHeader = "main-header";
+  const petsHeader = "pets-header";
```
По названию не понятно что это за хедер, лучше добавить в имя className -- `mainHeaderClassName`

hhhatemeee commented on src/pages/MainPage.jsx:
```diff
@@ -0,0 +1,235 @@
+import card from "../assets/icons/credit-card.svg";
+import { useState } from "react";
+
+function MovePage(path) {
+  window.location.href = path;
```
Т.к. мы используем react-router-dom, то вместо нативного window.location, можно использовать хук useNavigate

`const navigate = useNagivate()`

Используешь по такому же принципу, как и свою функцию: `navigate('/path')`

hhhatemeee commented on src/pages/Pets.jsx:
```diff
@@ -0,0 +1,52 @@
+import { useState } from "react";
+
+function Pets() {
+  const [cards, setCards] = useState([
+    { id: "slider-katrine", name: "Katrine", img: katrine },
+    { id: "slider-jennifer", name: "Jennifer", img: jennifer },
+    { id: "slider-woody", name: "Woody", img: woody },
+    { id: "slider-scarlett", name: "Scarlett", img: scarlett },
+    { id: "slider-sophia", name: "Sophia", img: sophia },
+    { id: "slider-timmy", name: "Timmy", img: timmy },
+    { id: "slider-charly", name: "Charly", img: charly },
+    { id: "slider-freddie", name: "Freddie", img: freddie },
+  ]);
```
Здесь можно обойтись обычным объектом с карточками, использовать хук useState тут нет надобности, т.к. в последствии ты не используешь setCards, массив остается неизменным

hhhatemeee commented on src/components/Slider.jsx:
```diff
@@ -0,0 +1,36 @@
+  };
+
+  const generatePopup = (id) => {
+    setPet(petsArray.find((n) => n.id == id));
```
метод .find возвращает найденный объект или undefined, на всякий случай нужно обработать случай, если вдруг у нас не будет найдет наш объект

hhhatemeee commented on src/App.jsx:
```diff
@@ -0,0 +1,22 @@
+      <Routes>
+        <Route path="/" element={<Layout />}>
+          <Route index element={<MainPage />} />
+          <Route path="pets" element={<Pets />} />
```
Создай отдельную директорию constants, где у тебя будут храниться файлы с константами. В данном случае 

`const ROUTES = { PETS: '/pets', MAIN: '/' }`

/_constants_ 
---_routes.js_
               
и т.д.

UnderSaga committed:
Изменения:
- Правки замечаний;
- Создание 3-ей страницы для создания/удаления животных;
- Реализация пагинации;
- Рефактор структуры проекта.

UnderSaga committed:
Изменения:
- Реализация функционала 3-ей страницы.

UnderSaga committed:
Изменения:
- Улучшено отображение модальных окон животных.

UnderSaga committed:
Изменения:
- Дополнительное улучшение модальных окон животных;
- Полный функционал слайдера.

UnderSaga committed:
Изменения:
- Поправлена работа кнопок пагинации.
